src/box/memtx_space.c

```diff
@@ -580,18 +585,19 @@ memtx_space_execute_upsert(struct space *space, struct txn *txn,
 		 * for the tuple.
 		 */
 		uint64_t column_mask = COLUMN_MASK_FULL;
+		size_t region_svp = region_used(&fiber()->gc);
 		const char *new_data =
 			xrow_upsert_execute(request->ops, request->ops_end,
 					    old_data, old_data + bsize,
 					    format, &new_size,
 					    request->index_base, false,
 					    &column_mask);
-		if (new_data == NULL)
-			return -1;
-
-		new_tuple =
-			space->format->vtab.tuple_new(format, new_data,
-						      new_data + new_size);
+		if (new_data != NULL)
+			new_tuple =
+				space->format->vtab.tuple_new(format, new_data,
+							      new_data +
+								new_size);
```

Comment:
Nit: The indentation looks bad. Why not reformat like this?

```C
		if (new_data != NULL) {
			new_tuple = space->format->vtab.tuple_new(
					format, new_data, new_data + new_size);
		}
```

Reply:
Done.